### ISSUE TITLE
Handle case where `Built` subclass passes the wrong generates for the builder

### ISSUE THREAD

**Issue by kevmoo** [google/built_value.dart#39]
``` dart
abstract class PubData implements Built<PubData, ChatBuilder> { 
...
```

I'd expect this to fail build – it works, but creates `PubDataBuilder`

**Comment by davidmorgan** [comment]
Thanks! PTAL:

https://github.com/google/built_value.dart/pull/43

**Comment by davidmorgan** [comment]
Fixed, will be in 0.1.6.
